gemma-teixido/pre-curso/tema-2/airlines-pro.js

```diff
@@ -0,0 +1,407 @@
+  );
+}
+
+function eliminarVuelos() {
+  newDisplayFlights();
+  let respuestaNumeroIdEliminar = prompt(
+    "Actualmente hay " + flights.length + " vuelos. Cual desea eliminar?"
+  );
+  respuestaNumeroIdEliminar *= 1;
+  flights.splice(respuestaNumeroIdEliminar, 1);
+  newDisplayFlights();
+}
+
+function questionUser() {
+  let questionUserPrice = prompt("Quiere buscar por precio? y/n");
+  if (questionUserPrice === "y") {
+    isUser();
+  } else if (questionUserPrice === "n") {
+    bye();
+  } else {
+    alert("Try again with please");
+    questionUser();
+  }
+}
+
+function isUser() {
+  let searchPriceFrom = prompt("Introduce un precio");
+  let searchCondition = prompt(
+    "- Quieres ver los vuelos mas caros que " +
+      searchPriceFrom +
+      " $ -> PRESIONE 1 \n- Mas baratos que " +
+      searchPriceFrom +
+      " $ -> PRESIONE 2 \n- Igual a " +
+      searchPriceFrom +
+      " $ -> PRESIONE 3 ?"
+  );
+  if (searchCondition === "1") {
+    priceGreaterThan(searchPriceFrom);
+  } else if (searchCondition === "2") {
+    priceLowerThan(searchPriceFrom);
+  } else if (searchCondition === "3") {
+    priceEqualThan(searchPriceFrom);
+  } else {
+    alert("Wrong number, try again please");
+    isUser();
+  }
+}
+
+function checkAnswerPriceUser(priceNewVuelo) {
+  if (!priceNewVuelo || priceNewVuelo.match(/[a-z]/i)) {
+    alert("Info no válida. Introduzca los datos otra vez, por favor.");
+    console.log("Info no válida. Introduzca los datos otra vez, por favor.");
+    isUser();
+  }
+}
+
+function priceGreaterThan(searchPriceFrom) {
+  let greaterThan = flights.filter((flights) => flights.cost > searchPriceFrom);
+  console.log(
+    "**************************************   GREATER THAN " +
+      searchPriceFrom +
+      " $ **********************************************"
+  );
+  for (let i = 0; i < greaterThan.length; i++) {
+    if (greaterThan[i].scale === false) {
+      console.log(
+        `El vuelo con origen -> ${greaterThan[i].from}, y destino -> ${greaterThan[i].to} tiene un coste de ${greaterThan[i].cost} $ y no realiza ninguna escala.`
+      );
+    } else {
+      console.log(
+        `El vuelo con origen -> ${greaterThan[i].from}, y destino -> ${greaterThan[i].to} tiene un coste de ${greaterThan[i].cost} $ y si que realiza escala.`
+      );
+    }
+  }
+  console.log(
+    "**************************************   GREATER THAN " +
+      searchPriceFrom +
+      " $ **********************************************"
+  );
+  console.log("Gracias por su compra, que tenga un buen dia! :)");
+}
+
+function priceLowerThan(searchPriceFrom) {
+  let lowerThan = flights.filter((flights) => flights.cost < searchPriceFrom);
+  console.log(
+    "**************************************   LOWER THAN " +
+      searchPriceFrom +
+      " $ **********************************************"
+  );
+  for (let i = 0; i < lowerThan.length; i++) {
```

Comment:
For such a simple for loop better use for..in or for..of instead. Apply to all for loop with this simplicity:

for (let i=0; i< something.length; i++)